### DISCUSSION TITLE
Advice for Using Inter for GNOME

### DISCUSSION THREAD

**Discussion by cassidyjames** [rsms/inter#761] [General]
Hey @rsms, I didn't know where the best place to reach out would be, but I figured an issue would be good so other GNOME contributors can follow along if it's helpful. :)

We've been evaluating [switching from Cantarell to Inter for the default GNOME UI typeface](https://gitlab.gnome.org/Teams/Design/initiatives/-/issues/152) for some time, and are finally moving forward. A few points came up that I was curious to get your opinion on as the maintainer—apologies if some of these take some explanation as I'm trying to synthesize comments from a handful of discussions with a bunch of different contributors:

### Baking in stylistic alternates

There is a lot of love for the disambiguated lower-case L in Cantarell, so we would be interested in baking `cv05` into the version in GNOME. Unfortunately we can't just set stylistic alternatives e.g. in CSS globally because downstreams like Ubuntu (or even end users, or users of GNOME apps on other desktops) may be using a different font, and we don't really want to special case styles based on which font is currently being used.

I see [OpenType Feature Freezer](https://twardoch.github.io/fonttools-opentype-feature-freezer/) recommended on the Inter site for this, but would it be more "proper" to build a variant from source? Some contributors felt like the feature freezer felt like a hack, but we _also_ don't want to end up in a case where we're maintaining an entire derivative font—we _just_ want to enable an existing feature in the font itself.

Ultimately I think we'd be building a derivative (whether with the feature freezer or from source) in CI anyway, so it's not hugely consequential _to me_ how that works—but I am curious to hear your thoughts or suggestions here.

### Metrics

Cantarell is a bit tighter width-wise, so some contributors are worried that switching to Inter will break some layouts or worst case ellipsize strings earlier. In testing, we find that reducing the default font size when switching to Inter (e.g. from 11 to 10) gets us closer to the metrics of Cantarell, but we were curious if you had any other suggestions off the top of your head here. One contributor was wondering if we can/should increase the default optical sizing that is baked into the font (again, because it's difficult to lean on CSS for this since we only want it to apply to this specific font, and ideally we would want it to carry through to other apps using other toolkits).

I don't see a way to do that in OpenType Feature Freezer, so perhaps that would necessitate us building from source, anyway?

### Naming

If we do ship a slightly-tweaked Inter in GNOME, what would you prefer we call it? E.g. we could go with something disconnected from Inter like "GNOME UI", or we could be more upfront about it being a variant of it with something like "Inter GNOME"; I can see branding reasons from your side to prefer one or the other, and I don't think anyone really cares what we call it on the GNOME side as it's not really user-facing. :)

---

Thank you so much for your tireless work on Inter; it truly has been a huge boon to open source projects everywhere to have a modern, quality, well-maintained typeface using all the fancy modern font features. :)

**Comment by cassidyjames** [comment]
For now we have gone with the name `GNOME UI`, but we'd be happy to change it if you prefer. We've set up CI using OpenType Feature Freezer here: https://gitlab.gnome.org/GNOME/gnome-ui-font

**Comment by jamiethecat** [comment]
Since we use Inter in the desktop's interface, which includes arbitrary strings such as generated passwords and URLs, we have had some trouble with the `calt` font feature, namely the sudden transformation of `x` to `×`. Using CSS to disable this feature is not an option as mentioned in the issue, is there a preferred route to take here? I can think of two ways to work around this from our side:
1. Use `pyftsubset` to exclude the feature from the font entirely
2. Patch `calt.fea` to exclude the `x` conversion, and build the font ourselves

**Comment by cassidyjames** [comment]
Another point that came up from the GNOME design team: there have been a few fixes in the main branch that have not been released, so a new point release would be very welcome so we can ship those fixes. :)

**Comment by cassidyjames** [comment]
I just realized there's a Discussions tab, and this would probably make more sense over there. 🤦🏻 I think repo admins are able to move it—if so, please feel free to do so!

**Comment by rsms** [comment]
Hello! GNOME UI sounds like a good name to me.

Excited to see this all happening!

> @jamiethecat "we have had some trouble with the calt font feature, namely the sudden transformation of x to ×"

Yeah this is a controversial transformation which I'm inclined to remove in future releases of Inter. While many designers I've talked with praises this feature, I think it may be causing more issues that the assist it provides.

> @jamiethecat "is there a preferred route to take here?"

~Until it changes (if it does) in mainline Inter, I would apply a patch to remove the problematic statements in `src/features/calt.fea` (like literally `patch -p1 < calt.path`) during your build process.~ Edit: removed in https://github.com/rsms/inter/commit/dee8b1228bb393309b048d565938d8178bf98613

> @cassidyjames "a new point release would be very welcome"

Working on it!

**Reply by kra-mo** [reply]
> While many designers I've talked with praises this feature, I think it may be causing more issues that the assist it provides.

Another example here is in Figma. The software doesn’t actually let you multiply numbers with the x character, only *, yet it looks like it does because of the transformation.

**Comment by rsms** [comment]
("Discussions" would have been better but let's just keep it going here for now :)
Also how random is it that we both commented on this just now?! Crazy

**Comment by rsms** [comment]
Just removed the NxN transformation from calt in https://github.com/rsms/inter/commit/dee8b1228bb393309b048d565938d8178bf98613

**Comment by rsms** [comment]
Turns out there's a "convert issue to discussion" function, so that was easy — now a discussion thread instead of an issue

**Reply by cassidyjames** [reply]
Thanks!

**Comment by kra-mo** [comment]
The question that still remains then is that of metrics. Is there a way we could reduce the default letter spacing baked into the font without building it ourselves? And if it’s only by building it, how could we go about doing it then?
